src/neml2/models/solid_mechanics/PerzynaPlasticFlowRate.cxx

```diff
@@ -32,42 +32,47 @@ ParameterSet
 PerzynaPlasticFlowRate::expected_params()
 {
   ParameterSet params = PlasticFlowRate::expected_params();
-  params.set<Real>("eta");
-  params.set<Real>("n");
+  params.set<Real>("reference_stress");
+  params.set<Real>("exponent");
   return params;
 }
 
 PerzynaPlasticFlowRate::PerzynaPlasticFlowRate(const ParameterSet & params)
   : PlasticFlowRate(params),
-    _eta(register_parameter("reference_flow_stress", Scalar(params.get<Real>("eta")))),
-    _n(register_parameter("flow_rate_exponent", Scalar(params.get<Real>("n"))))
+    _eta(register_parameter("eta", Scalar(params.get<Real>("reference_stress")), false)),
+    _n(register_parameter("n", Scalar(params.get<Real>("exponent")), false))
 {
 }
 
 void
-PerzynaPlasticFlowRate::set_value(LabeledVector in,
-                                  LabeledVector out,
-                                  LabeledMatrix * dout_din) const
+PerzynaPlasticFlowRate::set_value(const LabeledVector & in,
+                                  LabeledVector * out,
+                                  LabeledMatrix * dout_din,
+                                  LabeledTensor3D * d2out_din2) const
 {
+  const auto options = in.options();
+  const auto nbatch = in.batch_size();
+
   // Grab the yield function
   auto f = in.get<Scalar>(yield_function);
 
   // Compute the Perzyna approximation of the yield surface
-  Scalar Hf = torch::heaviside(f, torch::zeros_like(f));
```

Comment:
Define a nice heaviside function in the math utils as `torch::sign(x) + 1)/2`

Reply:
Done